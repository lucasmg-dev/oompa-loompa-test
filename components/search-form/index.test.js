import '@testing-library/jest-dom'
import { fireEvent } from '@testing-library/dom'

import { SearchForm } from './index'
import { render, screen } from '@testing-library/react'

const onKeyUp = jest.fn()

it('renders correctly', () => {
  const { asFragment } = render(<SearchForm handleFilter={onKeyUp} />)
  expect(asFragment()).toMatchSnapshot()
})

it('call filter handler', () => {
  render(<SearchForm handleFilter={onKeyUp} />)
  fireEvent.keyUp(screen.getByPlaceholderText('Search'))
  expect(onKeyUp).toHaveBeenCalledTimes(1)
})
