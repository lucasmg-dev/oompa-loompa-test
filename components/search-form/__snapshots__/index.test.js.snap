// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="searchNav"
  >
    <div
      class="wrapper"
    >
      <form
        class="searchForm"
      >
        <div
          class="searchFormGroup"
        >
          <input
            class="searchInput"
            placeholder="Search"
            type="text"
            value=""
          />
          <div
            class="searchIcon"
          >
            <img
              src=""
              width="15"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
